src/main/java/spoon/reflect/visitor/QueryVisitor.java

```diff
@@ -42,8 +46,8 @@ public QueryVisitor(Filter<T> filter) {
 		if (filter instanceof AbstractFilter) {
 			filteredType =  ((AbstractFilter<T>) filter).getType();
 		} else {
-			Class<?>[] params = RtHelper.getMethodParameterTypes(filter.getClass(), "matches", 1);
-			filteredType = (Class<T>) params[0];
+			Method method = RtHelper.getMethod(filter.getClass(), "matches", 1);
```

Comment:
You did not check here that `method` can be null as you did in `AbstractFilter`: it's on purpose because you assume it can never happens or is it missing?